beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -379,302 +254,175 @@ func (bs *Server) PublishBlockV2(w http.ResponseWriter, r *http.Request) {
-				http2.WriteError(w, errJson)
+				http2.HandleError(w, err.Error(), http.StatusBadRequest)
 				return
 			}
 			bs.proposeBlock(ctx, w, consensusBlock)
 			return
 		}
+		if versionHeader == version.String(version.Bellatrix) {
+			blockVersionError = fmt.Sprintf(": could not decode %s request body into consensus block: %v", version.String(version.Bellatrix), err.Error())
+		}
 	}
 	var altairBlock *shared.SignedBeaconBlockAltair
 	if err = unmarshalStrict(body, &altairBlock); err == nil {
-		if err = validate.Struct(altairBlock); err == nil {
-			consensusBlock, err := altairBlock.ToGeneric()
-			if err != nil {
-				errJson := &http2.DefaultErrorJson{
-					Message: "Could not decode request body into consensus block: " + err.Error(),
-					Code:    http.StatusBadRequest,
-				}
-				http2.WriteError(w, errJson)
-				return
-			}
+		consensusBlock, err := altairBlock.ToGeneric()
+		if err == nil {
 			if err = bs.validateBroadcast(ctx, r, consensusBlock); err != nil {
-				errJson := &http2.DefaultErrorJson{
-					Message: err.Error(),
-					Code:    http.StatusBadRequest,
-				}
-				http2.WriteError(w, errJson)
+				http2.HandleError(w, err.Error(), http.StatusBadRequest)
 				return
 			}
 			bs.proposeBlock(ctx, w, consensusBlock)
 			return
 		}
+		if versionHeader == version.String(version.Altair) {
+			blockVersionError = fmt.Sprintf(": could not decode %s request body into consensus block: %v", version.String(version.Altair), err.Error())
+		}
 	}
 	var phase0Block *shared.SignedBeaconBlock
 	if err = unmarshalStrict(body, &phase0Block); err == nil {
-		if err = validate.Struct(phase0Block); err == nil {
-			consensusBlock, err := phase0Block.ToGeneric()
-			if err != nil {
-				errJson := &http2.DefaultErrorJson{
-					Message: "Could not decode request body into consensus block: " + err.Error(),
-					Code:    http.StatusBadRequest,
-				}
-				http2.WriteError(w, errJson)
-				return
-			}
+		consensusBlock, err := phase0Block.ToGeneric()
+		if err == nil {
 			if err = bs.validateBroadcast(ctx, r, consensusBlock); err != nil {
-				errJson := &http2.DefaultErrorJson{
-					Message: err.Error(),
-					Code:    http.StatusBadRequest,
-				}
-				http2.WriteError(w, errJson)
+				http2.HandleError(w, err.Error(), http.StatusBadRequest)
 				return
 			}
 			bs.proposeBlock(ctx, w, consensusBlock)
 			return
 		}
+		if versionHeader == version.String(version.Phase0) {
+			blockVersionError = fmt.Sprintf(": Could not decode %s request body into consensus block: %v", version.String(version.Phase0), err.Error())
+		}
 	}
-
-	errJson := &http2.DefaultErrorJson{
-		Message: "Body does not represent a valid block type",
-		Code:    http.StatusBadRequest,
+	if versionHeader == "" {
+		blockVersionError = fmt.Sprintf(": please add the api header %s to see specific type errors", api.VersionHeader)
 	}
-	http2.WriteError(w, errJson)
+	http2.HandleError(w, "Body does not represent a valid block type"+blockVersionError, http.StatusBadRequest)
```

Comment:
I think it's more straightforward to include the `: ` in the first part of this message and remove it from the beginning of all `fmt.Sprintf` calls.